crates/config/src/lib.rs

```diff
@@ -640,7 +642,30 @@ impl Config {
         Self::from_provider(provider)
     }
 
+    /// Read `foundry.toml`, collect any unknown keys, and log a warning.
+    fn warn_unknown_keys() -> Result<(), ExtractConfigError> {
+        let path = Path::new(Self::FILE_NAME);
+
+        // if the file exists & is readable, deserialize and collect ignored keys
+        if let Ok(raw) = fs::read_to_string(path) {
+            let mut ignored = Vec::new();
+            let deserializer = toml::Deserializer::new(&raw);
+
+            // collect every ignored field name
+            let _: Result<Self, _> =
+                serde_ignored::deserialize(deserializer, |field| ignored.push(field.to_string()));
+
+            // if we found any, log a single warning with a list
+            if !ignored.is_empty() {
+                warn!("Found unknown config keys in {}: {}", Self::FILE_NAME, ignored.join(", "));
```

Comment:
this should use `sh_warn` and output message. We already have collect_warnings so should use that one

https://github.com/foundry-rs/foundry/blob/66fe662eea664d73c22a0753f5fb9fe78e2ec982/crates/config/src/providers/warnings.rs#L41

Would be great to address also the inline config, that is https://github.com/foundry-rs/foundry/issues/5371